infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/ExternalServerIngressBuilder.java

```diff
@@ -89,6 +90,14 @@ public ExternalServerIngressBuilder withMachineName(String machineName) {
     return this;
   }
 
+  public ExternalServerIngressBuilder withLabels(Map<String, String> labels) {
```

Comment:
from `withLabels` name, I would assume to use only given labels. However, this adds given labels to existing ones. There is very similar method `withAnnotations` that behaves differently. What do you think to rename this to somethink like `addLabels`?

Or is there even a reason why to `putAll` instead of replace everything?